cleanlab/rank.py

```diff
@@ -556,52 +555,87 @@ def get_confidence_weighted_entropy_for_each_label(
-      This is the "query set" of features for each datapoint which are used for nearest neighbor search.
+    features : np.ndarray
+      Feature array of shape ``(N, M)``, where N is the number of examples and M is the number of features used to represent each example.
+      All features should be numeric. For unstructured data (eg. images, text, categorical values, ...), you should provide
+      vector embeddings to represent each example (e.g. extracted from some pretrained neural network).
 
-    nbrs : sklearn.neighbors.NearestNeighbors
-      Instantiated NearestNeighbors class object that's been fitted on a dataset in the same feature space.
+    knn : sklearn.neighbors.NearestNeighbors, default = None
+      Instantiated ``NearestNeighbors`` class object that's been fitted on a dataset in the same feature space.
       Note that the distance metric and n_neighbors is specified when instantiating this class.
+      You can also pass in a subclass of ``sklearn.neighbors.NearestNeighbors`` which allows you to use faster
+      approximate neighbor libraries as long as you wrap them behind the same sklearn API.
+      If you specify ``knn`` here and wish to find outliers in the same data you already passed into ``knn.fit(features)``, you should specify ``features = None`` here if your ``knn.kneighbors(None)``
+      returns the distances to the datapoints it was ``fit()`` on.
+      If ``knn = None``, then by default ``knn = sklearn.neighbors.NearestNeighbors(n_neighbors=k, metric="cosine").fit(features)``
+
       See: https://scikit-learn.org/stable/modules/neighbors.html
 
-    k : int, default=None
-      Number of neighbors to use when calculating average distance to neighbors.
-      This value k needs to be less than or equal to max_k which is the n_neighbors used when fitting instantiated NearestNeighbors class object.
-      If k=None, then by default k=min(10, max_k) is used where max_k is extracted from the given nbrs.
+    k : int, default=10
+      Number of neighbors to use when calculating outlier score (average distance to neighbors). If an existing ``knn``
+      object is provided, you can still specify that outlier score should use a different value of ``k`` than originally
+      used in the ``knn``, as long as your specified value of ``k`` is smaller than the one originally used.
+      If k is larger, then by default ``k = n_neighbors`` where ``n_neighbors`` is defined in the ``knn``.
+      If k is not provided, then by default ``k = 10``.
+
+    t : int, default=1
+      Parameter controlling magnitude of similarity score transformation. Transformation is ``exp(-x*t)`` where ``x`` is the average distance to k-nearest neighbors.
 
     Returns
     -------
-    avg_nbrs_distances : np.array
-      Average distance to k-nearest neighbors for each datapoint which is used as a score for OOD detection.
+    avg_knn_distances : np.ndarray
```

Comment:
[nit] So I did this but I don't think it is intuitive to have an array called "outlier_scores" where higher score is less likeley to be an outlier. Something like "in_distribution_scores" seems better. Only raising this because this array name will be in the function description and may be confusing

Reply:
Renaming would probably require renaming entire function though